external/build/deprecation-calendar.js

```diff
@@ -0,0 +1,121 @@
+const {default: fetch} = require('node-fetch');
+const fs = require('fs').promises;
+const path = require('path');
+
+const DEPRECATION_ENDPOINT =
+  'https://chromestatus.com/api/v0/features?q=feature_type=3';
+
+const CHANNELS_ENDPOINT = 'https://chromestatus.com/api/v0/channels';
+
+/**
+ * Fetches all the deprecations from a list of features
+ *
+ * @return {promise} A promise that resolves in an array containing
+ * all the deprecations
+ */
+async function fetchDeprecations() {
+  let deprecations = [];
+  try {
+    const data = await fetch(DEPRECATION_ENDPOINT);
+
+    const deprecationsText = (await data.text()).substring(5); // Remove first 5 chars to have a correctly formatted JSON
+    const deprecationsJson = JSON.parse(deprecationsText);
+    deprecations = deprecationsJson?.features.sort(deprecationsSort);
+  } catch (error) {
+    console.error('Error fetching the deprecations', error);
+  }
+  return deprecations;
+}
+
+/**
+ * Fetches all channels info needed to get depracation
+ * dates starting from a Chrome version
+ *
+ * @param {number | undefined} start oldest version needed
+ * @param {number  | undefined} end newest version needed
+ * @return {promise} A promise that resolves in an array containing
+ * all the channel infos needed
+ */
+async function fetchChannels(start, end) {
+  let channels = [];
+  let queryString = '';
+  if (start !== undefined && end !== undefined) {
+    queryString = `?start=${start}&end=${end}`;
+  }
+
+  try {
+    const data = await fetch(`${CHANNELS_ENDPOINT}${queryString}`);
+
+    const channelsText = (await data.text()).substring(5); // Remove first 5 chars to have a correctly formatted JSON
+    channels = JSON.parse(channelsText);
+  } catch (error) {
+    console.error('Error fetching the channels', error);
+  }
```

Comment:
Consider refactoring this into a reusable `apiFetch` (or anything) method to deduplicate L19-25